### ISSUE TITLE
vscode ignores files.watcherExclude

### ISSUE THREAD

**Issue by mic-e** [microsoft/pylance-release#2974]
## Environment data

- Language Server version: 2022.6.30
- OS and version: Ubuntu 20.04 LTS
- Python version: 3.8.10 (default, Mar 15 2022, 12:22:08)
- Visual Studio Code version: 1.68.1

## Expected behavior

PyLance watches all files that are relevant to the project, but ignores files that are listed in the VSCode setting `files.watcherExclude`

## Actual behavior

PyLance ignores `files.watcherExclude`

## Description

This is a possible follow-up to #2914.

I work with a SSH remote host; on this host I am not able to adjust the maximum number of `inotify` watches.

`PyLance` immediately opens the maximum amount (8172) of file watches, using up all resources. Because of this, basic system commands such as `journalctl -f` stop working properly while VSCode is connected to the remote system.

When inspecting the `inotify` watches I can see that pylance is monitoring many files in `/usr/lib/python3.8`, `/usr/lib/python3/dist-packages`, `/opt/ros/noetic/lib/python3/dist-packages`.

I attempted to blacklist these folders through the `files.watcherExclude` setting by adding entries such as `/usr/lib/python3/**` but this had no effect.

**Comment by heejaechang** [comment]
@mic-e vscode should respect the option and exclude folders from watching. I also just tested it myself on windows and confirmed that it works as expected. if it didn't work on linux, It sounds like a bug in vscode file watcher implementation.

**Comment by heejaechang** [comment]
@mic-e it looks like github doesn't allow me to transfer this issue to vscode repo. can you open new issue on vscode repo (https://github.com/Microsoft/vscode)? vscode should respect the option, if not, vscode side needs to fix the bug.

**Comment by maciejmatczak** [comment]
That issue explicitly states that manually provided exclude folder list is ignored.

I am not really interested in workarounding aggressive file watch strategy (if I understand the case properly).

What is a root change that caused all that issue? Is it an update in Pylance?

**Comment by binh-vu** [comment]
I also had this issue on Centos 7, where I cannot control the maximum number of inotify watches.

I believe the mentioned issues ([microsoft/vscode-remote-release#6436](https://github.com/microsoft/vscode-remote-release/issues/6436), [microsoft/vscode#137872](https://github.com/microsoft/vscode/issues/137872)) are when they try to use glob pattern. For now on Linux, the vscode team recommended to use the full path or relative path to the workspace directory.

I did try with the full path:
```
"files.watcherExclude": {
    "${workspaceFolder}/.venv": true,
    "/home/user/anaconda3/lib/python3.9": true
}
```

Without activate pylance, I didn't have any error. As soon as I open a python file, I got file watcher error:
`[error] [File Watcher (parcel)] Inotify limit reached (ENOSPC) (path: /home/user/anaconda3/lib/python3.9)`. So it watches the excluded folder outside of the workspace.

This is related to the issue #2914 and [is said to be fixed](https://github.com/microsoft/pylance-release/issues/2914#issuecomment-1163777245) in version 2022.6.30. I tested on that version and the latest, and it still doesn't work.
